lighthouse/multi-cluster-service-selection.md

```diff
@@ -0,0 +1,71 @@
+# Multi-Cluster Service Selection Optimization - EP
+
+## Motivation
+
+While the world is moving towards micro-services application based architecture, more organizations combine multi-cluster and multi-cloud providers for redundancy, performance, management, and other reasons.
+We would like to optimize the total cost of a Kubernetes deployment in means of communication between multi-clusters, given that we could decide between different approaches. The three approaches possible, currently, are optimizing the full service chain of a single app/deployment, multiple apps/deployments, or a smart local optimization per cluster point of view, without building the full service chain.
+In this Enhancement Proposal, we shall introduce questions, ideas, and requirements to optimize the service selection mechanism of the Lighthouse Plugin.
+
+
+### Requirements
+1. Topology:
+    1. Selection Decision should be made locally, i.e per cluster - decentralized solution.
+    2. The solution must be adaptive for topology changes.
+1. Cost Parameters/Metrics:
+    1. Latency
+    2. RTT
+    3. Networking cost
+    4. Network congestion
+    5. CPU/RAM load on each container/pod
+    6. Service scale/autoscale (which actually will affect the load on the CPU and RAM)
+
+
+### Open Questions
+1. Topology:
+    1. What topology would we like to optimize?

```

Comment:
Yep a good question, in topology I meant what we can see from the deployment perspective (i.e single application deployment/multiple application deployment or indeed just a simple local and somehow smart decision from the cluster perspective without taking into account the whole application graph)... I only asked that because according to literature the more we "know" about the application graph the better optimization we can offer
And indeed like you mentioned in other comments, it stepping into the service mesh territory which we need to understand if we want to tackle - from the comments I can understand we don't want that.